src/android/aidl/birthday-service/birthday_service/aidl/com/example/birthdayservice/IBirthdayService.aidl

```diff
@@ -0,0 +1,35 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ANCHOR: IBirthdayService
+package com.example.birthdayservice;
+
+import com.example.birthdayservice.IBirthdayInfoProvider;
+
+// ANCHOR: with_info_provider
+/** Birthday service interface. */
+interface IBirthdayService {
+    /** Generate a Happy Birthday message. */
+    String wishHappyBirthday(String name, int years);
+
+    /** The same thing, but using more Binder features. */
```

Comment:
Probably worth being specific about which feature, in this case, callbacks?

Reply:
Yup, I've changed the wording here to be clearer, and the example has been simplified to only use the one binder feature (sending a binder object).